### PR TITLE
cpu/stm32: added optional RS485 module to uart periph

### PR DESCRIPTION
**Pull request by david-vankampen**
### Contribution description

added optional RS485 module to uart periph, to leverage DE pin capabilities

### Testing procedure

Updated a board that has a UART with the optional extra data, update the board's makefile to have `FEATURES_REQUIRED += periph_uart_rs485` and verify that the DE pin toggles as expected using a logic analyzer.

### CODE REVIEW

**Review by benpicco — COMMENTED** [review]
Hm it looks you forgot to commit something.
You are missing a 

    FEATURES_PROVIDED += periph_uart_rs485

in the cpu's `Makefile.features` and `HAS_PERIPH_UART_RS485` is not selected by anything.

Also is data enable really all it takes for RS485? What is it used for?

Maybe a `periph_uart_de` (like `periph_uart_hw_fc`) would be a better name.

**Comment by david-vankampen** [comment]
@benpicco from the reference manual: 
image
this allows the MCU to interact with an RS-485 transceiver like [this TI chip ](https://www.ti.com/document-viewer/THVD1439/datasheet/GUID-9D11057E-F990-4E6A-990B-E0FA0B3DAD9A#TITLE-SLLF79X3399) - the DE and RE inputs there can be tied to the DE line out of the MCU.  DE means driver enable, RE I believe means Read Enable - basically allows for a half-duplex connection to the RS-485 bus.  I will take a cut at adding the `FEATURES_PROVIDED` entry and `HAS_PERIPH_RS485` as requested.

**Comment by david-vankampen** [comment]
@benpicco with the updates I just pushed, if I run the following command on `tests/periph_uart`: 
`FEATURES_REQUIRED=periph_uart_rs485 make BOARD=nucleo-g070rb`, and then use `arm-none-eabi-objdump` to check for the conditional code, I see it: 
```
arm-none-eabi-objdump -S bin/nucleo-g070rb/tests_periph_uart.elf | grep de_pin
    if (uart_config[uart].de_pin != GPIO_UNDEF) {
        gpio_init(uart_config[uart].de_pin, GPIO_IN);
        gpio_init_af(uart_config[uart].de_pin, uart_config[uart].de_af);
```

whereas if I omit the `FEATURES_REQUIRED` variable from the call to make, I do not see those symbols included.  I believe this should address your questions.

**Comment by david-vankampen** [comment]
@benpicco checking in - is anything still needed here?

**Comment by fjmolinas** [comment]
@david-vankampen Kconfig needs an update to match the feature changes!

**Comment by david-vankampen** [comment]
@fjmolinas hows that?

File: cpu/stm32/Kconfig [hunk 1]
```diff
@@ -15,6 +15,7 @@ config CPU_STM32
     select HAS_PERIPH_RTT_OVERFLOW
     select HAS_PERIPH_UART_MODECFG
     select HAS_PERIPH_UART_NONBLOCKING
+    select HAS_PERIPH_UART_RS485
```
Comment (leandrolanzieri) [inline cpu/stm32/Kconfig:18]: Should only be selected by the specific families (like done in `Makefile.features`), please check the specific Kconfig files in `cpu/stm32/kconfigs/<family>/Kconfig`.

Comment (david-vankampen) [reply cpu/stm32/Kconfig]: @leandrolanzieri done.

**Pull request closed without merging**